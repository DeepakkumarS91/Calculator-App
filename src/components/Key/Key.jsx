import React from 'react';
import PropTypes from 'prop-types';
import './Key.css';
const Key = ({keyValue,keyAction,keyType}) => {
    return ( 
        <div className={`key-container ${keyType}`}
        onClick={()=>keyAction(keyValue)}>
<p className="key-value">
{keyValue}
</p>
        </div>
     );
}
Key.propTypes={
    keyValue:PropTypes.string.isRequired,
    keyType:PropTypes.string.isRequired,
    keyAction:PropTypes.func.isRequired
}

export default Key;