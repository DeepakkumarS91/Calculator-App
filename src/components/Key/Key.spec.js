import React from 'react';
import {shallow} from 'enzyme';
import Key from './Key';

describe('checks for Key component',()=>{
    let wrapper;
    beforeEach(()=>wrapper=shallow(<Key
    keyAction={jest.fn()}
    keyValue={''}
    keyType={''}/>));

    it('should render correctly',()=>expect(wrapper).toMatchSnapshot());

    it('checks for div',()=>{
        expect(wrapper.find('div').length).toEqual(1);
    })

    it('checks for the keyvalue props',()=>{
        wrapper.setProps({keyValue:'test'});
        expect(wrapper.text()).toEqual('test')
    })
})