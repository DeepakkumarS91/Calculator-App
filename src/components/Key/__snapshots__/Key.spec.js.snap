// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks for Key component should render correctly 1`] = `ShallowWrapper {}`;
